include/ada/unicode-inl.h

```diff
@@ -18,10 +18,31 @@
 namespace ada::unicode {
 ada_really_inline size_t percent_encode_index(const std::string_view input,
                                               const uint8_t character_set[]) {
-  return std::distance(
-      input.begin(), std::ranges::find_if(input, [character_set](const char c) {
-        return character_sets::bit_at(character_set, c);
-      }));
+  const char* data = input.data();
```

Comment:
I am not opposed to this change, @anonrig, but did you determine that it was worth doing? This seems significantly more complicated than what we had. It is fine if it is worth it...

Reply:
It is not noticable due to the complexity of Ada, but the instruction count should have been lower with the new implementation. Do you see different/worse results?